### PR TITLE
exercisse-3

### PR DESCRIPTION
**Pull request by aniekanokpokpo**

### CODE REVIEW

File: .env.test [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Flask server configuration.
+FLASK_APP=app
+FLASK_ENV=development
+FLASK_SKIP_DOTENV=True
+
+# Change the following values for local development.
+SECRET_KEY=secret-key
+KEY = 'test'
```
Comment (Jonesey13) [inline .env.test]: You don't need quotes here (also I'd forgo the spaces)

File: item.py [hunk 2]
```diff
@@ -1,10 +1,28 @@
+from datetime import date, datetime
```
Comment (Jonesey13) [inline item.py:1]: You don't want the chromedriver file above committed into source control.

File: tests/test_integration.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+    # Create the new app.
+    test_app = app.create_app()
+    # Use the app to create a test_client that can be used in our tests.
+    with test_app.test_client() as client:
+        yield client
+
+
+def test_index_page(requests_mock, client):
```
Comment (Jonesey13) [inline tests/test_integration.py:20]: `requests_mock` is not patched and therefore does not resolve (i.e. the test cannot be run)

This error blocks this review being passed.

File: tests/test_selenuim.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
+import os
+from threading import Thread
+
+import pytest
+from selenium import webdriver
```
Comment (Jonesey13) [inline tests/test_selenuim.py:5]: `selenium` is not a dependency listed in `requirements.txt`. This prevents the project being setup.

This also blocks this review from passing.

**Review by Jonesey13 — CHANGES_REQUESTED** [review]
Hi Aniekan. This still needs some more work to get this past review:

1.  The integration test is not runnable atm.
1.  There's a missing dependency `selenium`
1.  `README.md` does not document how to run your tests

File: item.py (hunk 2 above)
Comment (aniekanokpokpo) [reply item.py:1]: I'm not sure what you mean here @Jonesey13

File: tests/test_selenuim.py (hunk 4 above)
Comment (aniekanokpokpo) [reply tests/test_selenuim.py:5]: updated requirements file

File: tests/test_integration.py (hunk 3 above)
Comment (aniekanokpokpo) [reply tests/test_integration.py:20]: updated requirements file which should have `request_mock` dependency

File: .env.template [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# Flask server configuration.
+FLASK_APP=app
+FLASK_ENV=development
+
+# Change the following values for local development.
+SECRET_KEY=secret-key
+KEY = '62ce4878a6a08e0d91f50f1de7ef29be'
+TOKEN = 'fe926cd3836c9d391b11d756c4445b78da11e145d38d89c5ee133fa35ab5a5b7'
+LIST_ID = '5f516c55b4b1d670902e4e36'
+DONE_LIST_ID = '5f51742dc2cdc729993a8914'
```
Comment (hugh-emerson) [inline .env.template]: You've put your credentials in github! These values should all be blank in .env.template or `KEY=change me to your Trello account key` or similar.

Also, FYI the .env file is not Python. So technically you should not have those spaces around the `=` signs. The python-dotenv library can handle what you've written, but some parsers will fail

File: README.md [hunk 6]
```diff
@@ -39,4 +39,8 @@ Now visit [`http://localhost:5000/`](http://localhost:5000/) in your web browser
 The `.env` file is used by flask to set environment variables when running `flask run`. This enables things like developement mode (which also enables features like hot reloading when you make a file change).
 * There's also a [SECRET_KEY](https://flask.palletsprojects.com/en/1.1.x/config/#SECRET_KEY) variable which is used to encrypt the flask session cookie.
 
-When running `setup.sh`, the `.env.template` file will be copied to `.env` if the latter does not exist.
\ No newline at end of file
+When running `setup.sh`, the `.env.template` file will be copied to `.env` if the latter does not exist.
+
+how to run tests
+- ensure chromedriver is downloaded and added to system path (https://chromedriver.chromium.org/downloads)
```
Comment (hugh-emerson) [inline README.md:45]: You also actually need Chrome installed.

**Review by hugh-emerson — APPROVED** [review]
A couple of minor points, but looks good. Merge when you're ready.

File: .env.template (hunk 5 above)
Comment (aniekanokpokpo) [reply .env.template]: done both

**Pull request merged**